src/java.base/share/classes/sun/security/util/AbstractAlgorithmConstraints.java

```diff
@@ -70,14 +71,68 @@ static Set<String> getAlgorithms(String propertyName) {
         return algorithmsInPropertySet;
     }
 
+    // Allocate the immutable lists as needed. Overwriting is not a concern.
+    private static List<String> aliasEdDSA = null;
+    private static List<String> aliasEd25519 = null;
+    private static List<String> aliasXDH = null;
+    private static List<String> aliasX25519 = null;
+
+    /**
+     * getAlias() adds extra algorithm names to the String if matched.  Used by
+     * checkAlgorithm(), it returns additional names that may be on the
+     * DisabledAlgorithmConstraints list.
+     */
+    public static List<String> getAliases(String algorithm) {
+        return switch (algorithm) {
+            case "EdDSA" -> {
+                if (aliasEdDSA == null) {
+                    aliasEdDSA = List.of("EdDSA", "Ed25519", "Ed448");
+                }
+                yield aliasEdDSA;
+            }
+            case "Ed25519" -> {
+                if (aliasEd25519 == null) {
+                    aliasEd25519 = List.of("EdDSA", "Ed25519");
```

Comment:
Hmm. Should disabling Ed25519 also disable EdDSA? I can see the reverse, but isn't Ed25519 meant to be a specific parameter set for EdDSA?

Reply:
This is complicated by `KeyPairGenerator.getInstance("EdDSA")` returning an Ed25519 key

If someone were to check permits() with "EdDSA" the above code recognizes that "Ed25519" on the disabled algorithm list overlaps with "EdDSA".  This is the first test in the test coded included in the PR.

Reply:
Do we call `permits` before instantiating a `KeyPairGenerator`? What if people call `kpg.initialize(NPS.Ed448)` after the instantiation?

In reality, I think it depends on how many `permits` calls there are. Modern algorithms have the key same algorithm name and signature algorithm name. When a signature operation is carried out, do we check on both the signature algorithm and the key? It seems only checking on the key is enough. It's actually more precise, since you can get the exact parameter set name there. This is why I asked if the method is "never called on a family algorithm name". When checking a key, if we always call `permits` on the parameter set name, we get the precise result.

Reply:
`permits()` are used in situations for jdk[tls|certpath|jar].disabledAlgorithms, and the SSLAlgorithmConstraints.  It's not called for APIs like KPG, Signature, etc.

Reply:
That's what I meant. Suppose in TLS when you verify a signature and you call `permits` on both the signature algorithm name and the key used to init the signature, it's OK if only one fails.

Reply:
Yes, it should check both the signature algorithm and key algorithm in one permits call for signatures.